Update Docs

vedantmgoyal2009 opened the pull request:
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Are you working against an Issue?

-----


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/1471)

vedantmgoyal2009 commented:
@denelon

denelon commented:
Did you check the spell checking changes? They are still failing.

vedantmgoyal2009 commented:
> Did you check the spell checking changes? They are still failing.

I did run the commands but it doesn't seem to be fixed.

denelon commented:
OK, the "end goal" is to add the spelling words to "expect.txt" for "words" or to "allow.txt" for "technical terms or portions of a URL" that were caught (in alphabetical order).

vedantmgoyal2009 commented:
@denelon spell check successful.

denelon commented:
I don't think you intended to delete doc/ManifestSpecv0.1.md

vedantmgoyal2009 commented:
> I don't think you intended to delete doc/ManifestSpecv0.1.md

Yes, because I think it is outdated and not used after the 1.0 schema came out. Should I revert that commit?

denelon commented:
OK, since that specification is no longer supported.

lechacon commented on .github/actions/spelling/expect.txt:
```diff
@@ -384,3 +384,9 @@ XPLATSTR
 xsi
 Zanollo
 zy
+commande
```
Several of these come from  `doc/windows/package-manager/package/manifest.md`:
```
ShortDescription: Le nouveau terminal Windows, une expérience de ligne de commande à onglets pour Windows.
```

There is a pattern for this line in `.github/actions/spelling/patterns.txt`:
```
# doc/ManifestSpecv1.0.md
^ShortDescription: "Le nouveau.*"$
```

vedantmgoyal2009 replied:
But the bot was still commenting the words as unrecognized.

lechacon replied:
I think that's because the pattern doesn't match anymore without the quotes. If you remove the quotes from the pattern it would probably match the line again and we wouldn't need to list commande, une, rience, onglets:
```
^ShortDescription: Le nouveau.*$
```

vedantmgoyal2009 replied:
@lechacon It's failing again.

vedantmgoyal2009 commented:
@denelon

jedieaston commented on doc/windows/package-manager/package/manifest.md:
```diff
@@ -18,7 +16,6 @@ This article describes the contents of a package manifest for Windows Package Ma
 
 The YAML format was chosen for package manifests because of its relative ease of human readability and consistency with other Microsoft development tools. If you are not familiar with YAML syntax, you can learn the basics at [Learn YAML in Y Minutes](https://learnxinyminutes.com/docs/yaml/).
```
I feel like there should be some mention of [microsoft/winget-create](https://github.com/microsoft/winget-create) here. If someone doesn't know YAML, that might be an easier way for them to get started.

(You don't have to do it @vedantmgoyal2009, just something I noticed looking over this).

vedantmgoyal2009 replied:
I have added a section for using `wingetcreate` and the YamlCreate script. Thanks for the suggestion.

jedieaston reviewed (changes requested):
Can you convert that MP4 file to a gif? MP4s aren't autoplayed by GitHub (and probably not autoplayed by docs.microsoft.com either...)

denelon commented:
Great call out @jedieaston. I don't think the tool was quite ready when that was written.

denelon commented:
@vedantmgoyal2009 Do you want to add a blurb about being able to use the wingetcreate utility, or go ahead and get this PR approved and merged?

vedantmgoyal2009 commented:
> @vedantmgoyal2009 Do you want to add a blurb about being able to use the wingetcreate utility, or go ahead and get this PR approved and merged?

@denelon I have added a section about using the tools. Also, I have converted the `mp4` file to `gif`.

denelon commented:
@vedantmgoyal2009 the expect.txt still shows the new words at the bottom of the file rather than alphabetically inline.

vedantmgoyal2009 commented:
I think you pressed the wrong button. Should I create another PR with your suggestions?

denelon commented:
@vedantmgoyal2009. I'm not really a fan of huge PRs. Smaller ones are far easier to validate and accept. This way at least the bulk of what you did is updated, and we can continue making incremental improvements. Documentation is one of the most valuable things a contributor can make on a project. Thanks for all of the hard work!

Yes, please make another PR 😊
